src/slider/tp-slider.ts

```diff
@@ -445,6 +467,70 @@ export class TPSliderElement extends HTMLElement {
 		this.removeAttribute( 'resizing' );
 	}
 
+	/**
+	 * Set Default Settings.
+	 */
+	setDefaultSettings() {
+		// Loop through the setting keys and set the default settings.
+		this.settingKeys?.forEach( ( key: string ) => {
+			// Check if the key attribute exists and the attribute name is in the settingKeys array.
+			if ( this.getAttribute( key ) && this.settingKeys.includes( key ) ) {
+				// Set all the default settings keys, equal to the value of that attribute.
+				this.defaultSettings[ key ] = this.getAttribute( key ) || '';
+			}
+		} );
+	}
+
+	/**
+	 * Update responsive settings.
+	 */
+	updateResponsiveSettings() {
+		// Check if responsiveSettings exist.
+		if ( ! this.responsiveSettings.length ) {
+			// Early Return.
+			return;
+		}
+
+		// Step 1: Initialize matchFound.
+		let matchFound: boolean = false;
+
+		// Step 2: Loop through responsiveSettings and check if the media query is matched.
+		this.responsiveSettings.forEach( ( settings: TPSliderSettings ) => {
+			// Check if media query is matched.
+			if ( window.matchMedia( settings.media ).matches ) {
+				// If yes, loop through the settings at this media breakpoint.
+				for ( const settingKey in settings ) {
+					// Check if the setting key is not media.
+					if ( 'media' !== settingKey ) {
+						// Set those keys as attributes.
+						this.setAttribute( settingKey, settings[ settingKey ] );
+					}
+				}
+
+				// Set match found to true.
+				matchFound = true;
+			}
+		} );
+
+		// Step 3: If match is not found, set defaults.
+		if ( ! matchFound ) {
```

Comment:
@imranhsayed Lets remove this. Lets us mandate that all cases be explicitly added in responsive, even the default ones - this will prevent us from dealing with weird edge cases in the future - and also reduce the burden on this library